lldb/include/lldb/Host/ProcessLaunchInfo.h

```diff
@@ -118,7 +122,15 @@ class ProcessLaunchInfo : public ProcessInfo {
 
   bool MonitorProcess() const;
 
-  PseudoTerminal &GetPTY() { return *m_pty; }
+#ifdef _WIN32
+  PseudoConsole &GetPTY() const { return *m_pty; }
+
+  std::shared_ptr<PseudoConsole> GetPTYSP() const { return m_pty; }
+#else
+  PseudoTerminal &GetPTY() const { return *m_pty; }
+
+  std::shared_ptr<PseudoTerminal> GetPTYSP() const { return m_pty; }
+#endif
```

Comment:
Rather than doing this, can we do something slightly different and use a typealias? Something like this should work:

```c++
#if defined(_WIN32)
using PTY = PsuedoConsole;
#else
using PTY = PsuedoTerminal;
#endif

PTY &GetPTY() { return *m_pty; }
std::shared_ptr<PTY> GetPTYSP() const { return m_pty; }
```